### PR TITLE
VEIOSS-154 Implement Check Your Answers Section

### PR DESCRIPTION
**Pull request by hmrcvmu**

### CODE REVIEW

File: app/controllers/CheckYourAnswersController.scala [hunk 1]
```diff
@@ -103,6 +117,47 @@ class CheckYourAnswersController @Inject()(
...
+  }
+
+  def onSubmit(incompletePrompt: Boolean): Action[AnyContent] = cc.authAndGetData().async {
+    implicit request =>
+      registrationService.fromUserAnswers(request.userAnswers, request.vrn) match {
+        case Valid(registration) =>
+          registrationConnector.submitRegistration(registration).flatMap {
+            case Right(_) =>
+              auditService.audit(RegistrationAuditModel.build(RegistrationAuditType.CreateRegistration, registration, SubmissionResult.Success, request))
+              Future.successful(Ok) // TODO: replace with sendEmailConfirmation
```
Comment (hmrcvmu) [inline app/controllers/CheckYourAnswersController.scala:131]: Replace with `sendEmailConfirmation` once https://github.com/hmrc/ioss-registration-frontend/pull/22 is merged

File: app/pages/SavedProgressPage.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import play.api.mvc.Call
+
+case object SavedProgressPage extends QuestionPage[String] {
+
+  override def path: JsPath = JsPath \ toString
+
+  override def toString: String = "continueUrl"
+
+  override def route(waypoints: Waypoints): Call =
+    routes.SavedProgressController.onPageLoad("continueUrl") // TODO: check logic
```
Comment (hmrcvmu) [inline app/pages/SavedProgressPage.scala:30]: SavedProgressPage in OSS doesn't have a route method https://github.com/hmrc/one-stop-shop-registration-frontend/blob/8d25bc6080d13845bcdc4f4b29db0116926b5874/app/pages/SavedProgressPage.scala

File: app/services/EuTaxRegistrationValidations.scala [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        DataMissingError(TaxRegisteredInEuPage).invalidNec
+    }
+  }
+
+  private def processEuDetail(answers: UserAnswers, index: Index): ValidationResult[EuTaxRegistration] = {
+    answers.get(EuCountryPage(index)) match {
+      case Some(country) => getRegistrationWithFixedEstablishment(answers, country, index) // TODO: check logic here
+      case None => DataMissingError(EuCountryPage(index)).invalidNec
+    }
+  }
```
Comment (hmrcvmu) [inline app/services/EuTaxRegistrationValidations.scala:58]: Double check the logic. Here's the same method in OSS https://github.com/hmrc/one-stop-shop-registration-frontend/blob/8d25bc6080d13845bcdc4f4b29db0116926b5874/app/services/EuTaxRegistrationValidations.scala#L54-L67 , but we don't have `SellsGoodsToEUConsumersPage` in IOSS

File: test/base/SpecBase.scala [hunk 4]
```diff
@@ -70,6 +73,11 @@ trait SpecBase
 
   val userAnswersId: String = "12345-credId"
 
+  val completeUserAnswers: UserAnswers = basicUserAnswersWithVatInfo
```
Comment (pbyrne84) [inline test/base/SpecBase.scala:76]: Can this handle Websites now as that is a required field?

File: app/controllers/previousRegistrations/AddPreviousRegistrationController.scala [hunk 5]
```diff
@@ -53,29 +56,47 @@ class AddPreviousRegistrationController @Inject()(
...
-              Future.successful(BadRequest(view(formWithErrors, waypoints, previousRegistrations, canAddCountries))),
-
-            value =>
-              for {
-                updatedAnswers <- Future.fromTry(request.userAnswers.set(AddPreviousRegistrationPage(), value))
-                _ <- cc.sessionRepository.set(updatedAnswers)
-              } yield Redirect(AddPreviousRegistrationPage().navigate(waypoints, request.userAnswers, updatedAnswers).route)
-          )
+      withCompleteDataAsync[PreviousRegistrationDetailsWithOptionalVatNumber](
+        data = getAllIncompleteDeregisteredDetails,
```
[the remark this answers is not in the document]
Comment (hmrcvmu) [reply app/controllers/previousRegistrations/AddPreviousRegistrationController.scala]: Updated

File: app/controllers/previousRegistrations/AddPreviousRegistrationController.scala [hunk 6]
```diff
@@ -53,29 +56,47 @@ class AddPreviousRegistrationController @Inject()(
 
           val previousRegistrations = PreviousRegistrationSummary.row(request.userAnswers, Seq.empty, waypoints, AddPreviousRegistrationPage())
 
-          Future.successful(Ok(view(form, waypoints, previousRegistrations, canAddCountries)))
-
+          withCompleteDataAsync[PreviousRegistrationDetailsWithOptionalVatNumber](
+            data = getAllIncompleteDeregisteredDetails,
```
[the remark this answers is not in the document]
Comment (hmrcvmu) [reply app/controllers/previousRegistrations/AddPreviousRegistrationController.scala]: Updated

File: test/controllers/previousRegistrations/AddPreviousRegistrationControllerSpec.scala [hunk 7]
```diff
@@ -56,6 +58,7 @@ class AddPreviousRegistrationControllerSpec extends SpecBase with MockitoSugar {
       .set(PreviousEuCountryPage(Index(0)), Country.euCountries.head).success.value
       .set(PreviousSchemePage(Index(0), Index(0)), PreviousScheme.OSSU).success.value
 
+  private val mockRegistrationConnector: RegistrationConnector = mock[RegistrationConnector]
```
Comment (pbyrne84) [inline test/controllers/previousRegistrations/AddPreviousRegistrationControllerSpec.scala:61]: this is not used I think

```
ioss-registration-frontend/test/controllers/previousRegistrations/AddPreviousRegistrationControllerSpec.scala:61:15: private val mockRegistrationConnector in class AddPreviousRegistrationControllerSpec is never used
[warn]   private val mockRegistrationConnector: RegistrationConnector = mock[RegistrationConnector]

```

**Review by pbyrne84 — APPROVED** [review]
LGTM

**Pull request merged**